streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/GlobalStoreNode.java

```diff
@@ -23,22 +23,22 @@
 import org.apache.kafka.streams.state.KeyValueStore;
 import org.apache.kafka.streams.state.StoreBuilder;
 
-public class GlobalStoreNode extends StateStoreNode {
+public class GlobalStoreNode<K, V> extends StateStoreNode {
```

Comment:
Could we replace raw type by wildcard type for StateStoreNode? After all, the raw type is strongly discouraged...

Reply:
Not sure what you mean? The old code uses "raw types" and we just add proper generic typing to get type safety for `GlobalStoreNode` class.

Reply:
> The old code uses "raw types" 

StateStoreNode uses "raw types", right? Why not replacing it by wildcard as "raw types" is discouraged. Further, the generic type of StoreBuilder doesn't seem to be useful. For example, StoreBuilder#build is called by following code.
```java
        for (final StoreBuilder<?> storeBuilder : globalStateBuilders.values()) {
            globalStateStores.put(storeBuilder.name(), storeBuilder.build());
        }
```

```java
        public StateStore build() {
            return builder.build();
        }
```

Both of they don't take the generic type.

Also, StreamsBuilder doesn't even use the generic type of StoreBuilder. Not sure what purpose for generic type of StoreBuilder?

Reply:
Ack. We can improve this further.

Note, this PR does not fix everything yet -- it's better to do it incrementally to keep PRs small -- I will include a fix for this comment in this PR though.